### ISSUE TITLE
[api/websocket] Skip notifications with missing data when iterating on notification list

### ISSUE THREAD

**Issue by nemesisdesign** [openwisp/openwisp-notifications#50]
Follow up to https://github.com/openwisp/openwisp-notifications/pull/44#discussion_r443243617

- raise an exception (a custom one we create for this purpose)
- catching this exception in the API and websocket when iterating on the notification list
- in case the exception is catched, the rendering of the notification is just skipped.

**Comment by pandafy** [comment]
**TLDR; don't save malformed notifications.** 

I worked on the `email_subject` property of Notification, since tests were failing for #57 ,  I omitted the object from serializer if NotificationException is caught. 

You can see in this screenshot below that only **1** notification is displayed whose email subject was rendered properly. **Rest 11 were discarded.** 

![image](https://user-images.githubusercontent.com/32094433/86384718-c1cc2600-bcac-11ea-9416-79c98e57653e.png)

This achieves the goal of our issue. Sweet! :wink: 

But, this is what we observe on admin dashboard. 

![image](https://user-images.githubusercontent.com/32094433/86384857-f809a580-bcac-11ea-817f-3e4c19370b4b.png)

The JS widget shows only one notification. :heavy_check_mark: 
But the unread notification badge still shows **1 unread notification**. :frowning_face: 

Unread notifications are fetched from notification through the query:
 ```
User.notifications.unread().count()
```
Since it is fetched from database, we can not account for malformed notifications.

One way will be to check each notification for correctness while creating count. But, this is a dirty solution since it will waste computational time each time a page is reloaded. :nauseated_face: 

Since, we are planning to remove admin page completely, we can simply not save malformed notifications at all. :smile:
If we are not planning to show them, why to save them in the first place? 

But not saving the notifications at all means that we might me throwing away data regarding network event, :thinking: 

We can save users for loosing much of the data by sending a notification to super users that this Notification Type is not correct. And any notification using it are being discarded. This way, super users will be able to inform the maintainer of the server to fix that.

**Comment by NoumbissiValere** [comment]
@pandafy and @nemesisdesign i think we can eliminate malformed notifications by setting default values for those attributes which makes the notification object malformed. This will ensure that all notifications we save in the database are correct. 
What do we think :smile:

**Comment by pandafy** [comment]
@NoumbissiValere I see defining `Notification Type` as the only place for human errors which can lead to malformed notifications. Fixing the notification type can possibly correct all related notifications, thus making all stored notifications valid. We already removed possibility of having notifications for deleted related objects in #44 .

**Comment by NoumbissiValere** [comment]
> We

yes  @pandafy from `Notification Type`, we could raise an `improperlyConfigure` Exception or set defaults for possible settings which could lead to errors.

**Comment by pandafy** [comment]
I don't think we can check for error in `message` and `email_subject` key of Notification Type. Each notification type can have different related objects, like `device`, `metric`, etc. I don't think we can check them before hand. (Not universally)

We can surely set a fallback value(default).

Something like 

    f'notification.actor notification.verb notification.target'
